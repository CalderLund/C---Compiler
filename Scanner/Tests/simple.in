// Simple scanner example

int wain(int a, int b) {
  int x = 0;
  int y = 0;
  
  if (a >= b) {
    x = a + b;
    y = b;
  } 
  else {
    x = a;
    y = b - a;
  }
  if (x == 2) {
    x = x + 1;
  } else {}
  
  return a * b / x - y;
}
