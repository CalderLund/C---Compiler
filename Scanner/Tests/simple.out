INT int
WAIN wain
LPAREN (
INT int
ID a
COMMA ,
INT int
ID b
RPAREN )
LBRACE {
INT int
ID x
BECOMES =
NUM 0
SEMI ;
INT int
ID y
BECOMES =
NUM 0
SEMI ;
IF if
LPAREN (
ID a
GE >=
ID b
RPAREN )
LBRACE {
ID x
BECOMES =
ID a
PLUS +
ID b
SEMI ;
ID y
BECOMES =
ID b
SEMI ;
RBRACE }
ELSE else
LBRACE {
ID x
BECOMES =
ID a
SEMI ;
ID y
BECOMES =
ID b
MINUS -
ID a
SEMI ;
RBRACE }
IF if
LPAREN (
ID x
EQ ==
NUM 2
RPAREN )
LBRACE {
ID x
BECOMES =
ID x
PLUS +
NUM 1
SEMI ;
RBRACE }
ELSE else
LBRACE {
RBRACE }
RETURN return
ID a
STAR *
ID b
SLASH /
ID x
MINUS -
ID y
SEMI ;
RBRACE }
